app/scripts/views/base.js

```diff
@@ -166,9 +166,11 @@ function (Cocktail, _, Backbone, Raven, $, p, AuthErrors,
           if (! isUserAuthorized) {
             // user is not authorized, make them sign in.
             var err = AuthErrors.toError('SESSION_EXPIRED');
-
             self.navigate(self._reAuthPage(), {
-              error: err
+              error: err,
+              data: {
+                redirectTo: Backbone.history.fragment
```

Comment:
Reaching into Backbone.history likes this feels a bit gross and assumes we are using Backbone.history. Views already know about the router. The router has intimate knowledge of what page should currently be displayed. I propose you extend the router with a function `getCurrentPage` or similar. The router's `getCurrentPage` function would return the current page. Any time `router->navigate` is called, it would update what it believes is the current page, or just return `Backbone.history.fragment`. At least all `Backbone.history` access will be contained in one location at that point.